dhis-2/dhis-web/dhis-web-commons/src/main/java/org/hisp/dhis/util/TranslationUtils.java

```diff
@@ -59,13 +60,19 @@
         return translations;
     }
 
-    public static Map<String, String> convertTranslations( Set<ObjectTranslation> translations, String locale )
+    public static Map<String, String> convertTranslations( Set<ObjectTranslation> translations, Locale locale )
     {
+
+        if ( translations == null || locale == null )
+        {
+            return null;
+        }
+
         Map<String, String> translationMap = new Hashtable<>();
 
         for ( ObjectTranslation translation : translations )
         {
-            if ( translation.getProperty() != null && translation.getValue() != null && translation.getLocale().equals( locale ))
```

Comment:
I mean comparing translation.getLocale() to locale.getLanguage() - is that correct? What if we have country?